src/main/kotlin/org/opensearch/replication/task/index/IndexReplicationTask.kt

```diff
@@ -646,6 +645,12 @@ open class IndexReplicationTask(id: Long, type: String, action: String, descript
          * it continues to be called even after the task is completed.
          */
         clusterService.removeListener(this)
+        val replicationStateParams = getReplicationStateParamsForIndex(clusterService, followerIndexName)
+        if (replicationStateParams?.get(REPLICATION_LAST_KNOWN_OVERALL_STATE) != ReplicationOverallState.PAUSED.name) {
+            log.info("Cleaning up retention lease")
+            val replMetadata = replicationMetadataManager.getIndexReplicationMetadata(this.followerIndexName)
+            retentionLeaseHelper.attemptRemoveRetentionLease(clusterService, replMetadata, this.followerIndexName)
+        }
```

Comment:
Does this re-introduce earlier problem? If the task is relocated onto a different node, would we lose the lease?

Reply:
Yes. Removed it now.